### PR TITLE
[Manual Search] Log multi-epi as snatched

### PR DESCRIPTION
**Pull request by fernandog**
Fixes https://github.com/pymedusa/SickRage/issues/229

### CODE REVIEW

**Comment by fernandog** [comment]
why Im getting:

```
 File "/home/osmc/SickRage/sickbeard/webserve.py", line 1425, in pickManualSnatch
        search_result = SearchResult(provider=provider, show=show_obj, url=sql_return['url'], episodes=ep_objs, quality=sql_return['quality'], name=sql_return['name'], size=sql_return['size'], seeders=sql_return['seeders'], leechers=sql_return['leechers'], release_group=sql_return['release_group'], version=sql_return['version'])
TypeError: __init__() got an unexpected keyword argument 'seeders'
```

if searchResult has seeders?

https://github.com/pymedusa/SickRage/blob/develop/sickbeard/classes.py#L35-L65

**Comment by labrys** [comment]
It's because `SearchResult`'s `__init__` only takes `episodes` as a parameter.  You need to modify the `__init__` to accept other arguments, or create the object without it and access the created objects' attributes to set the values.

**Comment by fernandog** [comment]
@labrys what about now ?

```
    ep_queue_item = search_queue.ManualSnatchQueueItem(search_result)
TypeError: __init__() takes exactly 3 arguments (2 given)

```

Old code wasn't passing `segment`.

**Comment by fernandog** [comment]
@labrys 

```
 File "/home/osmc/SickRage/sickbeard/search_queue.py", line 126, in add_item
    elif isinstance(item, (ManualSearchQueueItem, ManualSnatchQueueItem, FailedQueueItem)) and not self.is_ep_in_queue(item.segment):
AttributeError: 'ManualSnatchQueueItem' object has no attribute 'segment'

```

File: sickbeard/manual_snatch.py [hunk 1]
```diff
@@ -125,26 +125,27 @@ def collectEpisodesFromSearchThread(show):
     # Queued Searches
     searchstatus = SEARCH_STATUS_QUEUED
     for searchThread in sickbeard.searchQueueScheduler.action.get_all_ep_from_queue(show):
-        episodes += getEpisodes(searchThread, searchstatus)
+        if isinstance(searchThread, sickbeard.search_queue.ManualSearchQueueItem):
```
Comment (p0psicles) [inline sickbeard/manual_snatch.py]: why are you checking this? get_all_ep_from_queue() only returns ManualSearchQueueItem. So now your validating redundant?

File: sickbeard/manual_snatch.py [hunk 2]
```diff
@@ -125,26 +125,27 @@ def collectEpisodesFromSearchThread(show):
...
     # Running Searches
     searchstatus = SEARCH_STATUS_SEARCHING
     if sickbeard.searchQueueScheduler.action.is_manualsearch_in_progress():
         searchThread = sickbeard.searchQueueScheduler.action.currentItem
 
-        if searchThread.success:
-            searchstatus = SEARCH_STATUS_FINISHED
-
-        episodes += getEpisodes(searchThread, searchstatus)
+        if isinstance(searchThread, sickbeard.search_queue.ManualSearchQueueItem):
```
Comment (p0psicles) [inline sickbeard/manual_snatch.py]: same here

File: sickbeard/manual_snatch.py [hunk 3]
```diff
@@ -125,26 +125,27 @@ def collectEpisodesFromSearchThread(show):
...
 
     # Finished Searches
     searchstatus = SEARCH_STATUS_FINISHED
     for searchThread in sickbeard.search_queue.MANUAL_SEARCH_HISTORY:
-        if show is not None:
-            if not str(searchThread.show.indexerid) == show:
-                continue
-
-        if isinstance(searchThread, (sickbeard.search_queue.ManualSearchQueueItem, sickbeard.search_queue.ManualSnatchQueueItem)):
+        if isinstance(searchThread, sickbeard.search_queue.ManualSearchQueueItem):
```
Comment (p0psicles) [inline sickbeard/manual_snatch.py]: same here

**Comment by p0psicles** [comment]
Really @fernandog, I don't get it. You ask me for help? But in the mean time your doing all kinds of "fixups" that are just wrong. I'm done with this PR.

File: sickbeard/webserve.py [hunk 4]
```diff
@@ -1411,27 +1413,38 @@ def pickManualSnatch(self, show=None, season=None, episode=None, provider=None,
...
+        search_result.show = show_obj
+        search_result.url = sql_return['url']
+        search_result.quality = int(sql_return['quality'])
+        search_result.name = sql_return['name']
+        search_result.size = int(sql_return['size'])
+        search_result.seeders = int(sql_return['seeders'])
+        search_result.leechers = int(sql_return['leechers'])
+        search_result.release_group = sql_return['release_group']
+        search_result.version = int(sql_return['version'])
+        search_result.resultType = search_result.provider.provider_type
```
Comment (fernandog) [inline sickbeard/webserve.py]: @p0psicles is this still ok?

**Comment by fernandog** [comment]
@labrys @OmgImAlexis @p0psicles need aprove

**Comment by duramato** [comment]
Can say this fixes it, and works ofc lol

**Comment by fernandog** [comment]
@p0psicles 
to handle Season Packs i think we should do something similar to this. But I didn't understood the logic

https://github.com/pymedusa/SickRage/blob/develop/sickrage/providers/GenericProvider.py#L288-L317

**Comment by p0psicles** [comment]
@fernandog, I agree. Tough we really need to create a better looking snatch selection icon.

Ps. Could you make sure this get's merged, before merging this into dev? https://github.com/pymedusa/SickRage/pull/245

**Pull request merged**